docs/rst/tutorials/micro/custom_types/custom_types.rst

```diff
@@ -0,0 +1,159 @@
+
+    For creating custom types in a :ref:`Build System Component <tutorials_micro_build_system_components>`, check the instructions in each components' repository. For example `Renesas e2 Studio Readme <https://github.com/micro-ROS/micro_ros_renesas2estudio_component#adding-custom-packages-to-the-micro-ros-build>`_ or `ST Micro ST Cube IDE/MX <https://github.com/micro-ROS/micro_ros_stm32cubemx_utils#adding-custom-packages>`_
+
+Simple types
+------------
+
+Once the micro-ROS workspace is created, navigate to ``firmware/mcu_ws`` and create a new package for the custom messages:
+
+.. code-block:: bash
+
+  cd firmware/mcu_ws
+  ros2 pkg create --build-type ament_cmake my_custom_message
+  cd my_custom_message
+  mkdir msg
+  touch msg/MyCustomMessage.msg
+
+In the generated ``CMakeLists.txt`` file the following lines shall be added just before ``ament_package()`` line:
+
+.. code-block:: text
+
+  ...
+  find_package(rosidl_default_generators REQUIRED)
+
+  rosidl_generate_interfaces(${PROJECT_NAME}
+    "msg/MyCustomMessage.msg"
+   )
+  ...
+
+In the generated ``package.xml`` file the following lines shall be added:
+
+.. code-block:: xml
+
+  ...
+  <build_depend>rosidl_default_generators</build_depend>
+  <exec_depend>rosidl_default_runtime</exec_depend>
+  <member_of_group>rosidl_interface_packages</member_of_group>
+  ...
+
+The content of the ``msg/MyCustomMessage.msg`` file contains the message definition. For example:
+
+.. code-block:: text
+
+  bool bool_test
+  byte byte_test
+  char char_test
+  float32 float32_test
+  float64 double_test
+  int8 int8_test
+  uint8 uint8_test
+  int16 int16_test
+  uint16 uint16_test
+  int32 int32_test
+  uint32 uint32_test
+  int64 int64_test
+  uint64 uint64_test
+
+.. note::
+
+    Same approach can be taken with ``.srv`` and ``.action`` files for creating types for micro-ROS :ref:`services<tutorials_micro_user_api_services>` and :ref:`actions <tutorials_micro_user_api_actions>`.
+
+Now, the micro-ROS workspace workspace can be built as usual. As explained in `Getting started micro-ROS <https://docs.vulcanexus.org/en/latest/rst/tutorials/micro/getting_started/getting_started.html>`_, the ``ros2 run micro_ros_setup build_firmware.sh`` command will build all packages located inside ``mcu_ws``. In the micro-ROS application code the new message type can be used as usual:
+
+.. code-block:: c
+
+  #include <my_custom_message/msg/my_custom_message.h>
+
+  ...
+
+  my_custom_message__msg__MyCustomMessage msg = {0};
+
+  msg.byte_test = 3;
+  msg.uint32_test = 42;
+
+  ...
+
+  rclc_publisher_init_default(&publisher, &node, ROSIDL_GET_MSG_TYPE_SUPPORT(my_custom_message, msg, MyCustomMessage), "my_custom_publisher");
+  rcl_publish(&publisher, &msg, NULL);
+
+  ...
+
+
+Further information can be found in :ref:`Implementing custom interfaces <SinglePkgInterface>`.
```

Comment:
I would move this to the start of the page

Reply:
eeff374fe8912d337d54725b469b98eb3b623c77